### ISSUE TITLE
Bit shifting operators

### ISSUE THREAD

**Issue by Disclosure** [takagi/cl-cuda#85]
I am trying to apply a bit shifting **shr** operator in a do loop that I found in _built-in.lisp_ (line 127) like so:

```
(defkernel reduce-array (void ((in float*) (out float*)))
  (let ((x thread-idx-x))
    (with-shared-memory ((acc float 64))
      (do ((j 32 (shr j 1)))
	  ((< j 1))
	(when (< x j)
	  (set (aref acc x) (+ (aref acc x)
			       (aref acc (+ x j)))))
	(syncthreads))
      (when (= x 0)
	(set (aref out 0) (aref acc 0))))))
```

However in a Ubuntu 16.04 sbcl I am getting a function undefined error:

```
The function SHR is undefined.
   [Condition of type SIMPLE-ERROR]
```

The symbol appears to be exported in the _built-in.lisp_ file but it seems like it cannot be found for some reason, when one tries to list all autocomplete symbols in slime, the shr symbol does not appear to be there:

![capture](https://cloud.githubusercontent.com/assets/8075703/22183724/0a9c1e88-e0bc-11e6-9ae4-203438a9611d.PNG)

The CL ASH function does not appear (expectedly) to work inside the kernel definition either, so how do I go about applying bit shifting in defkernel? Please excuse me if I am asking something entirely obvious, I am quite rusty on my Lisp skills after all these years.

**Comment by takagi** [comment]
Thanks for reporting. I will check it.

**Comment by guicho271828** [comment]
Not entirely unrelated, but when i tried AVM it complained that coerce was not found.

**Comment by takagi** [comment]
@guicho271828 Yes, it is reported in https://github.com/takagi/avm/issues/44 . I need to take care of it too.
